manticore/wasm/module_structure.py

```diff
@@ -0,0 +1,280 @@
+import typing
+from dataclasses import dataclass
+from .types import (
+    TypeIdx,
+    TableIdx,
+    FuncIdx,
+    GlobalIdx,
+    MemIdx,
+    Byte,
+    ValType,
+    FunctionType,
+    TableType,
+    MemoryType,
+    GlobalType,
+    LimitType,
+    Name,
+    WASMExpression,
+    convert_instructions,
+    I32,
+    I64,
+    F32,
+    F64,
+)
+from .runtime_structure import (
+    Store,
+    ModuleInstance,
+    FuncAddr,
+    TableAddr,
+    MemAddr,
+    GlobalAddr,
+    FuncInst,
+    TableInst,
+    MemInst,
+    GlobalInst,
+    Value,
+)
+from wasm import decode_module, Section
+from wasm.wasmtypes import (
+    SEC_TYPE,
+    SEC_IMPORT,
+    SEC_FUNCTION,
+    SEC_TABLE,
+    SEC_MEMORY,
+    SEC_GLOBAL,
+    SEC_EXPORT,
+    SEC_START,
+    SEC_ELEMENT,
+    SEC_CODE,
+    SEC_DATA,
+)
+
+ExportDesc = typing.Union[FuncIdx, TableIdx, MemIdx, GlobalIdx]
+ImportDesc = typing.Union[TypeIdx, TableType, MemoryType, GlobalType]
+
+
+@dataclass
+class Function:
+    type: TypeIdx
+    locals: typing.List[ValType]
+    body: WASMExpression
+
+    def allocate(self, store: Store, module: ModuleInstance) -> FuncAddr:
+        a = FuncAddr(len(store.funcs))
+        store.funcs.append(FuncInst(module.types[self.type], module, self))
+        return a
+
+
+@dataclass
+class Table:
+    type: TableType
+
+    def allocate(self, store: Store, table_type: TableType) -> TableAddr:
+        a = TableAddr(len(store.tables))
+        store.tables.append(
+            TableInst([None for _i in range(table_type.limits.min)], table_type.limits.max)
+        )
+        return a
+
+
+@dataclass
+class Memory:
+    type: MemoryType
+
+    def allocate(self, store: Store, mem_type: MemoryType) -> MemAddr:
+        a = MemAddr(len(store.mems))
+        store.mems.append(
+            MemInst(
+                [
+                    0x0 for _i in range(mem_type.min * (64 * 1024))
+                ],  # TODO - these should be symbolic, right?
```

Comment:
TODO: yes;  them, envvars (if wasm engines use them) and stdin

Reply:
Also: just `[0x0] * mem_type.min * (64 * 1024)`

Btw why `64*1024`?

Reply:
64*1024 is the standard WASM memory page size